src/index/TextBlockWriter.cpp

```diff
@@ -0,0 +1,118 @@
+#include "TextBlockWriter.h"
+
+#include "index/TextIndexReadWrite.h"
+
+template <typename WordTextVecView, typename EntityTextVecView>
+void TextBlockWriter<WordTextVecView, EntityTextVecView>::
+    calculateAndWriteTextBlocks(size_t nofWordPostingsInTextBlock) {
+  AD_CONTRACT_CHECK(
+      nofWordPostingsInTextBlock > 0,
+      "Number of word postings in text block has to be larger than zero.");
+  for (auto&& chunk :
+       ::ranges::views::chunk(wordTextVecView_, nofWordPostingsInTextBlock)) {
+    for (const auto& row : chunk) {
+      currentWordTextVecWordIndex_ = row[0].getWordVocabIndex();
+      addWordPosting(row);
+    }
+    addBlock();
+  }
+}
+
+// _____________________________________________________________________________
+template <typename WordTextVecView, typename EntityTextVecView>
+void TextBlockWriter<WordTextVecView, EntityTextVecView>::addBlock() {
+  // Add all co-occurring entities to entityPostings_
+  addEntityPostingsUpToWordIndex();
```

Comment:
This function can return the `entityPostings` directly, then you don't need a member variable for this.